### DISCUSSION TITLE
Brainstorming

### DISCUSSION THREAD

**Discussion by Cadienvan** [Il-Libro-Open-Source/book#1] [Salotto]
Apro questa discussione per fare brainstorming sul progetto e saggiare la volontà iniziale di partecipazione.

Condividete idee di ogni tipo, suggerimenti, spunti, link a progetti simili, foto di gattini, qualsiasi cosa!

**Comment by Cadienvan** [comment]
Apro le danze con il primo punto.

**Di che si parla?**

Io pensavo a _due_ macro-temi che credo siano molto attuali e di interesse al giorno d'oggi e che permettono al mondo dev di avere una competenza che arricchirebbe di molto la qualità del contenuto:
- Un racconto distopico sull'Intelligenza Artificiale: Sì, ce ne sono tanti, ma quanti contengono informazioni reali e corrette e ragionano **veramente** sulle implicazioni che l'AI avrà sulle nostre vite da qui a qualche decennio? Vorrei creare un racconto verosimile, non un altro "L'AI stermina la razza umana perché capisce che non gli serve".
- La storia dell'Open Source, dalla sua nascita a oggi, condita da situazioni specifiche di persone che nel corso degli anni hanno partecipato al mondo Open Source: Case history di progetti open source diventati standard moderni, persone di spicco della community che raccontano il loro punto di vista e le loro storie più particolari, cose così.

**Reply by teschiopol** [reply]
Condivido il secondo punto. Alcune domande, vissute in prima persona, come spunto sono:

- Come inizio nell'open source (paradossalmente questo libro potrebbe essere la risposta)?
- Voglio e creo un progetto source, quando dovrei condividerlo? Perché dopo una prima analisi/idea o perché meglio una volta finita la beta?
- In quanto "creatore" come dovrei comportarmi, adotto gitflow come regole dal punto di vista tecnico?
- Eventuali sponsor, come divido? "Anzianità nel progetto", creatore, numero di pull request, ci sono delle "regole"?
- Quali sono i progetti open source che potrei prendere come riferimento?

Personalmente, mantenendo un taglio più tecnico e meno filosofico, lo stenderei come dei passi pratici (10 step per essere un buon contributor) e suddiviso poi come hai detto tu con case history comprensivi di nascita, sviluppo, adesso e futuro del progetto (una cronistoria tipo podcast One More Time)

**Reply by Balastrong** [reply]
Piacerebbe anche a me parlare di Open Source in questo libro, molto "meta" (un libro open source sull'open source :D)
Lascio un po' di considerazioni, tanto siamo in fase divergente/brainstorming.

Una parte di situazioni ed esperienze personali è sicuramente molto di valore, ma vedo strano come possa funzionare che A racconta la sua esperienza e B apre una pull request per modificare delle cose. Ci sta eh, magari typo o mettere enfasi su determinati aspetti, però boh l'ho pensata e volevo scrivere ste tre righe.

Per la parte tecnica su come essere di aiuto (contributor, maintainer, sponsor) più che un libro è un manuale... che magari va bene lo stesso! Troppa roba ci sarebbe da scrivere.

Vogliamo raccontare una storia per intrattenere il lettore, vogliamo condividere valore, vogliamo ispirare qualcuno? Insomma, qual è l'obiettivo di questo libro? Sicuramente capire dove vogliamo arrivare ci può aiutare a definire struttura e contenuti.

Ah, diamo per scontato che sia in italiano?

**Reply by KevinValmo** [reply]
Si potrebbero prevedere delle cartelle per lingua.
Si prende come riferimento una lingua e la scrittura viene fatta su quella.
Può essere una idea...

**Reply by NoriSte** [reply]
Uhm provo a pensare... cosa i partecipanti a questo progetto hanno da raccontare? Bene o male le "loro" storie, che in genere sono "come dev, devi superare questo ostacolo e io ci sono riuscito cosí", oppure "sono riuscito a guadagnare 3x e lavorare per aziende come X e Y grazie a questo". L'obbiettivo non é tanto dire "io ho fatto cosí" ma "questa cosa (tipo fare networking) é importante perché... e tizio, caio e sempronio lo hanno fatto in questo modo".

In questo modo ne esce un libro dove ogni argomento avrebbe un'intro sulle quale sarebbe facile andare d'accordo con gli altri contributor... e poi l'argomento viene poi elaborato con come i singoli contributor lo mettono in pratica.

Il risultato sarebbe una sorta di manuale contenente guide ed esempi su un tot di argomenti che possono essere utilissimi a tutti i livelli, junior, middle e senior!

Tipo mi vengono in mente i seguenti temi che spesso tratto in varia discussioni qua e lá:
1. Personal branding
2. Overcommunication
3. Lavoro in asincrono
4. Importanza di [networking e reputation, non solo skill](https://twitter.com/ossia/status/1079942716480344065?s=19)
5. Sviluppo OSS (non sono un esperto qui)


> Un racconto dispotico sull'Intelligenza Artificiale: Sì, ce ne sono tanti, ma quanti contengono informazioni reali e corrette e ragionano veramente sulle implicazioni che l'AI avrà sulle nostre vite da qui a qualche decennio? Vorrei creare un racconto verosimile, non un altro "L'AI stermina la razza umana perché capisce che non gli serve".

Mi fermerei a spiegare grossomodo come funzionano gli strumenti di AI e simili, pro e contro, cosa fanno bene e cosa no, sfatare alcuni miti e stop. Non credo che un "racconto sull'impatto" uscirebbe bene, non siamo scrittori e sarebbe troppo soggettivo... chi dovrebbve leggerlo e perché?


> La storia dell'Open Source, dalla sua nascita a oggi, condita da situazioni specifiche di persone che nel corso degli anni hanno partecipato al mondo Open Source: Case history di progetti open source diventati standard moderni, persone di spicco della community che raccontano il loro punto di vista e le loro storie più particolari, cose così.

Toglierei la parte di "storia del'OSS" e piuttosto mi soffermerei sulle storie, che é anche in linea con quello che dicevo sopra 😊

**Reply by pietrolerro** [reply]
Un minimo di storia la metterei, da Stallmann e la FSF.

**Reply by pietrolerro** [reply]
direi di cominciare in italiano vista la maggioranza di italian speakers e poi tradurlo una volta arrivati ad una beta funzionante

**Reply by dinb79** [reply]
Mettendo insieme alcune di queste idee, associando il taglio pratico, si potrebbe fare una specie di compendio di buone pratiche e esempi reali o realistici di come applicarle, ma soprattutto descriverne poi i vantaggi o gli svantaggi (ad esempio il tempo e le energie risparmiate).
Ad esempio, dei topic spesso ignorati nei primi anni di lavoro sono
- intelligenza emotiva (specialmente applicata nel mondo IT)
- smart working (quello vero), applicazioni di metodologie innovative (non per forza solo agile)
- comunicazione (specialmente col personale anziano: non dimentichiamo che in italia la maggior parte delle aziende non sono moderne, né tantomeno lo sono i loro dipendenti)
- analisi
- architettura (soprattutto in termini di strategia a lungo termine)
- documentazione efficace (vs documentare con documenti da centinaia di pagine)

**Reply by ghost** [reply]
Per quanto riguarda il primo punto, analizziamo le possibili implicazioni di sistemi AI sempre più avanzati ospitati su eventuali computer quantistici dalla potenza di calcolo praticamente illimitata. Me ne vengono in mente giusto un paio :
- Che venga data fin troppa fiducia a tali sistemi, come già accade, con conseguente svalutazione (ed in certi casi sostituzione) del lavoro umano;
- Come ulteriore conseguenza, una polarizzazione sempre più eccessiva delle ricchezze e del potere economico della popolazione globale.

Volendo rimanere negativi, è molto plausibile che tutto questo possa contribuire in modo non indifferente al collasso della società umana per come la conosciamo.

**Comment by pietrolerro** [comment]
qualche spunto: 
- Mi piacerebbe dare un taglio il più pratico possibile, alternando la narrativa ad esempi concreti. 
- Aggiungerei che tanto materiale dato in training alla AI è preso da sorgenti pubbliche (copilot legge da github, la nuova policy di google, ecc.)
- Porterei anche dei casi di roba "grossa" che è diventata open (mi viene in mente CUPS, ma non vorrei sbagliarmi, approfondirò)
- una overview delle licenze
- indicazioni pratiche per chi vuole cominciare
- modelli che possono anche portare un guadagno ("open" non vuol dire sempre "gratis")

grazie @Cadienvan, bella idea davvero! :)

**Comment by pietrolerro** [comment]
in realtà tutta questi discussione è sugli argomenti, con le PR definiamo il prompt e poi scrive tutto chatgpt /s

**Comment by GuidoPenta** [comment]
io posso contribuire coi gattini

**Comment by ilmanzo** [comment]
scrivere un racconto a piu' mani e' abbastanza complicato, specie tra sconosciuti; piuttosto ci starebbe una raccolta di racconti brevi. Se il libro e gli argomenti sono totalmente open, ognuno scrive un pezzo che gli pare e poi si fa il collage :) 
Magari stabiliamo alcuni paletti, tipo numero minimo/massimo di battute e un tema generale che puo' essere la tecnologia, la storia dell'informatica (sotto forma di racconto/diario personale) o i gattini di Guido.

**Reply by pietrolerro** [reply]
a pro di questo vorrei linkare un racconto carino: https://disconnessi.org/racconti/quando-i-sistemisti-domineranno-la-terra.html
e, per chi li ha letti, due libri "the phoenix project" e "the unicorn project".

**Reply by Cadienvan** [reply]
I project sono un must :)

**Comment by Cadienvan** [comment]
Mi sembrano tutti degli spunti interessanti e stra-pieni di grandiose idee!

Lasciamo decantare un po' questa cosa per qualche giorno, se la cosa vi interessa condividetela con amici e colleghi e magari a inizio settimana prossima facciamo una prima survey per mettere giù qualche paletto concreto. Che ne pensate?

**Reply by pietrolerro** [reply]
ha molto senso. magari definisci un end date per il brainstorming (magari tra un paio di gg), per dare il tempo a chi è interessato di partecipare, ma anche per avere una separazione netta tra le fasi di progetto :)

**Reply by Cadienvan** [reply]
Absolutely, pensavo a lunedì per lasciare tempo agli weekendari. 🤣

**Comment by MaSTERmIKK** [comment]
Ottima idea, anch'io do i miei due centesimi, penso che per iniziare istituirei delle "cose rules" nel senso di canoni specifici per il confronto e la proposta di idee in modo da incanalarle meglio nel evoluzione del progetto.

**Reply by Cadienvan** [reply]
Sicuramente sarà uno dei primissimi step!

**Comment by dinb79** [comment]
> * Un racconto dispotico sull'Intelligenza Artificiale:

Ma intendi proprio **dispotico**, o distopico? 
In ogni caso, il "racconto" fa molto sci-fi, che a me piace, ma non lo vedo compatibile con la raccolta di consigli, buone pratiche, esempi.
A meno che non si usino dei mini-racconti per introdurre i vari capitoli, tipo appunto una storia distopica sull'AI per il capito in cui si parla di AI, un altro _racconto distopico in cui c'è una pandemia mondiale che spinge i governi a introdurre lo smart working, e questo smart working mette in evidenza quanto il sistema sia poco preparato ad accoglierlo_, come introduzione al capito dello smart working, ecc ecc ;)

**Reply by Cadienvan** [reply]
Carina l'idea, comunque era un typo ahah

**Comment by crixo** [comment]
Il mio suggerimento e' di includere anche esperienze personali legate al mondo open source ovvero a come ci si e' approcciati e avvicinati. 
Attraverso l'accaduto di ciascuno, altri potrebbero trarre suggerimenti e idee su come e perche' partecipare alla community e ai progetti open source.

**Reply by Cadienvan** [reply]
Mi sembra una grande idea!

**Comment by Cadienvan** [comment]
Ragazzi, è aperto il poll (#2).
@buggyzap @NoriSte @crixo @ilmanzo @teschiopol @Balastrong @KevinValmo @pietrolerro @dinb79 @Cb040719 @MaSTERmIKK @GuidoPenta 

Lunedì spingo anche su LinkedIn, voi intanto fatemi sapere la vostra e condividete il poll con tutti!

**Comment by mirkomancin** [comment]
Scrivere qualcosa sulla open Innovation e legarla a temi come IoT e Realtà aumentata?

**Reply by Cadienvan** [reply]
Ciao @mirkomancin grazie mille per il feedback!

Se mi tiri giù due righe di riassunto di come lo vedi lo aggiungo alla survey!

**Comment by Cadienvan** [comment]
Avendo appena "concluso" le votazioni per il tema, direi che possiamo chiudere questa discussione!
